### PR TITLE
Revert `CountBy` return type to `KeyValuePair<,>`

### LINKED ISSUE #490: Revert `CountBy` return type to `KeyValuePair<,>`
Per https://github.com/dotnet/runtime/issues/77716#issuecomment-1664448577, the approved API for `CountBy` returns `IEnumerable<KeyValuePair<int, TSource>>` instead of `IEnumerable<(int, TSource)>`. SuperLinq should revert the return type to match the corelib approved API to encourage code-compatibility between SuperLinq and corelib.

### PR DESCRIPTION
**Pull request by tcortega**
Resolves #490

### CODE REVIEW

File: Tests/SuperLinq.Test/CountByTest.cs [hunk 1]
```diff
@@ -78,9 +78,9 @@ public void CountByWithSomeNullKeys()
...
 		ss.CountBy(SuperEnumerable.Identity)
 			.AssertSequenceEqual(
-				("foo", 2),
-				(null, 4),
-				("bar", 2),
-				("baz", 2));
+				KeyValuePair.Create("foo", 2),
+				KeyValuePair.Create((string)null!, 4),
+				KeyValuePair.Create("bar", 2),
+				KeyValuePair.Create("baz", 2));
```
Comment (viceroypenguin) [inline Tests/SuperLinq.Test/CountByTest.cs:84]:
```suggestion
				KeyValuePair.Create((string?)"foo", 2),
				KeyValuePair.Create((string?)null, 4),
				KeyValuePair.Create((string?)"bar", 2),
				KeyValuePair.Create((string?)"baz", 2));
```

File: Tests/SuperLinq.Async.Test/CountByTest.cs [hunk 2]
```diff
@@ -83,9 +83,9 @@ public async Task CountByWithSomeNullKeys()
...
 		await ss.CountBy(SuperEnumerable.Identity)
 			.AssertSequenceEqual(
-				("foo", 2),
-				(default, 4),
-				("bar", 2),
-				("baz", 2));
+				KeyValuePair.Create("foo", 2),
+				KeyValuePair.Create((string)null!, 4),
+				KeyValuePair.Create("bar", 2),
+				KeyValuePair.Create("baz", 2));
```
Comment (viceroypenguin) [inline Tests/SuperLinq.Async.Test/CountByTest.cs:89]:
```suggestion
				KeyValuePair.Create((string?)"foo", 2),
				KeyValuePair.Create((string?)null, 4),
				KeyValuePair.Create((string?)"bar", 2),
				KeyValuePair.Create((string?)"baz", 2));
```

**Review by viceroypenguin — COMMENTED** [review]
Looks good so far! Couple minor code changes. I would like this against v6 instead of master, because this is a 6.0 breaking change. As such, it'll need some conflict resolving in the unshipped.txt files.

**Comment by tcortega** [comment]
> Looks good so far! Couple minor code changes. I would like this against v6 instead of master, because this is a 6.0 breaking change. As such, it'll need some conflict resolving in the unshipped.txt files.

Done

**Review by viceroypenguin — APPROVED** [review]

**Comment by viceroypenguin** [comment]
Thanks @tcortega!

**Pull request merged**